### DISCUSSION TITLE
Problem in my code

### DISCUSSION THREAD

**Discussion by MrPRAINGLE** [discordjs/discord.js#6912] [Q&A] [answered]
const Discord = require("discord.js");
const Client = new Discord.Client({
    intents:[
        Discord.Intents.FLAGS.GUILDS,
        Discord.Intents.FLAGS.GUILD_MESSAGES,
        Discord.Intents.FLAGS.DIRECT_MESSAGES
    ]
});
const textecancel = "Envoyez ``cancel`` pour annuler la commande."
const { Permissons } = require("discord.js");
const prefix = "!";
const embedhelp = new Discord.MessageEmbed()
    .setTitle("<:PraingloHelp:903244322248458240>  Liste des commandes <:PraingloHelp:903244322248458240>")
    .setDescription("Vous trouverez ci-dessous toutes les commandes utilisables dans ce serveur !")
    .setColor("#658807")
    .addField("Configuration", "> ``help`` Envoie la liste des commandes disponibles\n> ``helpmp`` Envoie la liste des commandes disponibles en MP\n> ``userinfo`` Affiche la lise des informations à propos d'un membre (beta)");
const BlaguesAPI = require("blagues-api")
const embedclear1 = new Discord.MessageEmbed()
    .setTitle("Combien de messages voulez-vous supprimer ?")
    .setDescription(textecancel)
    .setColor("#ff0000")
const embednombreX = new Discord.MessageEmbed()
    .setTitle("Ceci n'est pas un nombre, veuillez réessayer !")
    .setDescription(textecancel)
    .setColor("#ff0000")
const embedclearX = new Discord.MessageEmbed()
    .setTitle("Tu ne peux pas faire ça !")
    .setDescription("*Il faut avoir la permission ``Gérer les messages`` pour utiliser cette commande.")
    .setColor("#ff0000")
const embedcancel = new Discord.MessageEmbed()
    .setTitle("🚫 Commande annulée !")
    .setColor("#DD2E44")

Client.on("ready", () => {
    console.log("Bot opérationnel ! ✌️");
});

Client.on("messageCreate", message => {
    if(message.content === prefix + "help"){
        message.delete()
        message.channel.send({ embeds: [embedhelp]})
    };

    if(message.content === prefix + "helpmp"){
        message.delete()
        message.author.send({ embeds: [embedhelp]})
        message.channel.send("@" + message.author.tag + ", je vous ai envoyé la liste des commandes en MP.")
    };

    if(message.content === prefix + "ui"){
        message.delete()
        const embedui = new Discord.MessageEmbed()
        .setThumbnail(message.author.avatarURL())
        .setAuthor("Informations à propos de " + message.author.tag, message.author.avatarURL())
        .setColor("#658807")
        .setDescription("Voici ce que je sais à propos de ``" + message.author.tag + "``.")
        message.channel.send({ embeds: [embedui]})
    };

    if(message.content === prefix + "userinfo"){
        message.delete()
        const embedui = new Discord.MessageEmbed()
        .setThumbnail(message.author.avatarURL())
        .setAuthor("Informations à propos de " + message.author.username, message.author.avatarURL())
        .setColor("#658807")
        .setDescription("Voici ce que je sais à propos de ``" + message.author.tag + "``.")
        message.channel.send({ embeds: [embedui]})
    };
    
    if(message.content.startsWith(prefix + "clear")){
        message.delete()
        if(message.author.Permissons = "MANAGE_MESSAGES"){
            let arg = message.content.trim().split(/ +/g)
            if(!arg[1]){
                message.channel.send({ embeds: [embedclear1]})
                await Client.on("messageCreate", message => {
                    if(message.content === Number){
                        let arg = message.content
                        const embedclearfinal = new Discord.MessageEmbed()
                        .addField("Suppression des messages", "``" + arg[1] + " messages ont bien été supprimés.``")
                        .setColor("#658807")
                        await message.channel.bulkDelete(arg[1])
                        await message.channel.send({ embeds: [embedclearfinal]}).then(message=>message.delete({timeout:"6000"}));
                    }else if(message.content === "clear"){
                        message.channel.send({ embeds: [embednombreX]})
                        return;
                    }else{
                        await message.channel.send({ embeds: [embedcancel]});
                    }
                }
            }
        }
    };
});

Here's my code. I have a little problem with the 4° } from the end, it is underlined with red... Can you help me please ?

**Comment by vvito7** [comment, marked as answer]
hi, i found the issue, you forgot to close the `await Client.on("messageCreate", message => { code });`
insteat you did just `await Client.on("messageCreate", message => { code }`

![image](https://user-images.githubusercontent.com/79790819/139555383-03c74412-88e5-4f45-a94a-f062bdf60f75.png)

hope i helped you :)

**Comment by MrPRAINGLE** [comment]
Thanks bro, it worked ! (Incredible that someone answered me so fast 😃)

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐
